src/components/Login/LoginContainer.test.tsx

```diff
@@ -0,0 +1,308 @@
+const loginSubmitOTPButtonId = "login-submit-otp-button";
+const scanButtonId = "base-button";
+const phoneInputId = "login-phone-number-input";
+const OTPInputId = "login-otp-input";
+const countryCodeInputId = "login-country-code-input";
+
+jest.mock("../../utils/errorTracking");
+const mockCaptureException = jest.fn();
+(Sentry.captureException as jest.Mock).mockImplementation(mockCaptureException);
+
+const mockDeepLink = jest.fn();
+jest.spyOn(Linking, "parseInitialURLAsync").mockImplementation(mockDeepLink);
+
+const mockRequestOTP = jest.spyOn(auth, "requestOTP");
+const mockValidateOTP = jest.spyOn(auth, "validateOTP");
+
+describe("LoginContainer", () => {
+  afterEach(() => {
+    mockCaptureException.mockReset();
+    mockDeepLink.mockReset();
+    mockRequestOTP.mockReset();
+    mockValidateOTP.mockReset();
+    cleanup();
+  });
+
+  it("should render LoginScanCard", () => {
+    expect.assertions(1);
+    const { getByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+  });
+
+  it("should render LoginMobileNumberCard if login endpoint + key in deep link", async () => {
+    expect.assertions(2);
+    mockDeepLink.mockImplementation(() =>
+      Promise.resolve({
+        queryParams: { key, endpoint },
+      })
+    );
+
+    const { getByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    await waitFor(() => {
+      expect(getByTestId(loginMobileViewId)).not.toBeNull();
+    });
+  });
+
+  it("should render Id scanner", () => {
+    expect.assertions(3);
+    const { getByTestId, queryByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    expect(getByTestId(barcodeScannerId)).not.toBeNull();
+    expect(queryByTestId(loadingViewId)).toBeNull();
+  });
+
+  it("valid QR detected and should render LoginMobileNumberCard", () => {
+    expect.assertions(4);
+    const { getByTestId, queryByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    expect(getByTestId(barcodeScannerId)).not.toBeNull();
+    expect(queryByTestId(loadingViewId)).toBeNull();
+
+    fireEvent(getByTestId(barcodeScannerId), "onBarCodeScanned", {
+      nativeEvent: { data: `{"key": "${key}","endpoint": "${endpoint}"}` },
+    });
+    expect(getByTestId(loginMobileViewId)).not.toBeNull();
```

Comment:
same observation here: mobile number view is not null regardless of whether we fire the `onBarCodeScanned` event or not